_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1066,6 +1074,89 @@ private function _process_onboarding( $data ) {
 			: join( ', ', $error );
 	}
 
+	/**
+	 * Add or update Business Address widget.
+	 *
+	 * @param array $address Array of business address fields.
+	 *
+	 * @return WP_Error|true True if the data was saved correctly.
+	*/
+	static function handle_business_address( $address ) {
+		$first_sidebar = Jetpack_Widgets::get_first_sidebar();
+
+		$widgets_module_active = Jetpack::is_module_active( 'widgets' );
+		if ( ! $widgets_module_active ) {
+			$widgets_module_active = Jetpack::activate_module( 'widgets', false, false );
+		}
+		if ( ! $widgets_module_active ) {
+			return new WP_Error( 'module_activation_failed', 'Failed to activate module.', 400 );
+		}
+
+		if ( $first_sidebar ) {
+			$title = isset( $address['name'] ) ? sanitize_text_field( $address['name'] ) : '';
+			$street = isset( $address['street'] ) ? sanitize_text_field( $address['street'] )
+			: '';
+			$city = isset( $address['city'] ) ? sanitize_text_field( $address['city'] ) : '';
+			$state = isset( $address['state'] ) ? sanitize_text_field( $address['state'] ) : '';
+			$zip = isset( $address['zip'] ) ? sanitize_text_field( $address['zip'] ) : '';
+
+			$full_address = implode( ' ', array_filter( array( $street, $city, $state, $zip ) ) );
+
+			$widget_options = array(
+				'title'   => $title,
+				'address' => $full_address,
+				'phone'   => '',
+				'hours'   => '',
+				'showmap' => false,
+				'email' => ''
+			);
+
+			$widget_inserted = '';
+			$widget_updated = '';
```

Comment:
Why do we need 2 separate variables for handling insert and update if we return the same error message?